### PR TITLE
Refine TiDB features document

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Refine TiDB features document:

- Abbreviate "Experimental" as "E" to save table space.
- Delete DMR version info before v6.5.
- Add a note to inform users that even if a feature is GA in a DMR version, they are not recommended to use that DMR version for the feature.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12863
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: basic-features.md [hunk 1]
```diff
@@ -6,224 +6,359 @@ aliases: ['/docs/dev/basic-features/','/tidb/dev/experimental-features-4.0/']
 
 # TiDB Features
 
-This document lists the features supported in each TiDB version.
+This document lists the features supported in different TiDB versions, including [Long-Term Support (LTS)](/releases/versioning.md#long-term-support-releases) versions and [Development Milestone Release (DMR)](/releases/versioning.md#development-milestone-releases) versions after the latest LTS version.
 
 > **Note:**
 >
-> - Y: the feature is generally available (GA) and can be used in production environments.
+> PingCAP does not provide patch releases (including bug fixes and security vulnerability fixes) for DMR versions. If you encounter any bugs in a DMR version, they will be fixed in the subsequent releases. It is recommended to use the [latest LTS version](https://docs.pingcap.com/tidb/stable) unless you have special needs.
```
Comment (lilin90) [inline basic-features.md:13]: Try to keep wording consistent with that in the **Important** module on https://docs.pingcap.com/tidb/v6.4.
```suggestion
> PingCAP does not provide patch releases for DMR versions. Any bugs will be fixed in future releases. For general purposes, it is recommended to use the [latest LTS version](https://docs.pingcap.com/tidb/stable).
```

File: basic-features.md [hunk 2]
```diff
@@ -6,224 +6,359 @@ aliases: ['/docs/dev/basic-features/','/tidb/dev/experimental-features-4.0/']
...
+    <td>Y</td>
+    <td>Y</td>
+    <td>Y</td>
+    <td>Y</td>
+    <td>Y</td>
+    <td>Y</td>
+    <td>Y</td>
+  </tr>
+</tbody>
+</table>
```
Comment (TomShawn) [inline basic-features.md:364]:
```suggestion
```

**Review by lilin90 — APPROVED** [review]
LGTM

File: basic-features.md [hunk 3]
```diff
@@ -6,224 +6,228 @@ aliases: ['/docs/dev/basic-features/','/tidb/dev/experimental-features-4.0/']
...
+| ------------------------------------------------------------ | :--: | :--: | ------------ | :----------: | :----------: | :----------: | :----------: | :----------: |
+| [Expression indexes](/sql-statements/sql-statement-create-index.md#expression-index) [^2] | Y | E | E | E | E | E | E ||
+| [Columnar storage (TiFlash)](/tiflash/tiflash-overview.md)   | Y | Y | Y            |      Y       |      Y       |      Y       |      Y       |      Y       |
+| [Use FastScan to accelerate queries in OLAP scenarios](/develop/dev-guide-use-fastscan.md) | E | N | N | N | N | N | N | N |
+| [RocksDB engine](/storage-engine/rocksdb-overview.md)        | Y | Y | Y            |      Y       |      Y       |      Y       |      Y       |      Y       |
+| [Titan plugin](/storage-engine/titan-overview.md)            | Y | Y | Y            |      Y       |      Y       |      Y       |      Y       |      Y       |
+| [Titan Level Merge](/storage-engine/titan-configuration.md#level-merge-E)   |  E   |  E    |    E     |    E     |    E     |    E     |    E     |    E     |
```
Comment (TomShawn) [inline basic-features.md:54]:
```suggestion
| [Titan Level Merge](/storage-engine/titan-configuration.md#level-merge-experimental)   |  E   |  E    |    E     |    E     |    E     |    E     |    E     |    E     |
```

**Pull request merged**